src/views/collection-view/index.tsx

```diff
@@ -1,29 +1,108 @@
 import Link from 'next/link'
+import { TutorialLite as ClientTutorialLite } from 'lib/learn-client/types'
+import useCurrentPath from 'hooks/use-current-path'
+import { useCurrentProduct } from 'contexts'
 import SidebarSidecarLayout from 'layouts/sidebar-sidecar'
 import CoreDevDotLayout from 'layouts/core-dev-dot-layout'
-import { TutorialLite as ClientTutorialLite } from 'lib/learn-client/types'
-import ProductCollectionsSidebar from 'components/tutorials-sidebar/compositions/product-collections-sidebar'
-import { getTutorialSlug } from './helpers'
+import {
+  generateProductLandingSidebarNavData,
+  generateTopLevelSidebarNavData,
+} from 'components/sidebar/helpers'
+import TutorialsSidebar, {
+  HorizontalRule,
+  SectionList,
+  SectionTitle,
+} from 'components/tutorials-sidebar'
+import { CollectionCategorySidebarSection, getTutorialSlug } from './helpers'
 import { CollectionPageProps } from './server'
 import CollectionMeta from './components/collection-meta'
 
 function CollectionView({
   collection,
-  product,
   layoutProps,
 }: CollectionPageProps): React.ReactElement {
-  const { name, slug, description, shortName, tutorials } = collection
+  const currentProduct = useCurrentProduct()
+  const currentPath = useCurrentPath({ excludeHash: true, excludeSearch: true })
+  const { name, slug, description, tutorials } = collection
+
+  // TODO: refactor, very long
+  const sidebarNavDataLevels = [
```

Comment:
Perhaps we could do some of this data generation in `getStaticProps` and some of the base boilerplate could be shared between views? It's hard for me to see how _similar_ these are between each view. 

Or what are your plans for refactoring this?

Reply:
🗒️ Ended up refactoring both `generateTopLevelSidebarNavData` and `generateProductLandingSidebarNavData` to return more data.